### ISSUE TITLE
Support SiTU activation function

### ISSUE THREAD

**Issue by pmilojevicTT** [tenstorrent/tt-metal#51350]
Kimi K3 uses Sigmoid Tanh Unit (SiTU) activation function, while our current compute kernel support only SiLU and SwiGLU-OAI activation functions.

This is current PyTorch imeplemntation

```
# Register Moonshot-specific activation functions
class SituAndMul(nn.Module):
    """
    SituAndMul activation: beta * tanh(gate / beta) * sigmoid(gate) * up
    When linear_beta is set, up is also transformed by linear_beta * tanh(up / linear_beta).
    """

    def __init__(self, beta: float = 1.0, linear_beta: float | None = None):
        super().__init__()
        self.beta = beta
        self.linear_beta = linear_beta

    def forward(self, x: torch.Tensor) -> torch.Tensor:
        d = x.shape[-1] // 2
        gate = x[..., :d].to(torch.float32)
        up = x[..., d:].to(torch.float32)
        situ_a = self.beta * torch.tanh(gate / self.beta) * torch.sigmoid(gate)
        if self.linear_beta is not None:
            up = self.linear_beta * torch.tanh(up / self.linear_beta)
        return (situ_a * up).to(x.dtype)
```

Both self.beta and self.linear_beta are used in Kimi K3 so both activation functions for GATE and UP are needed.

**Comment by pmilojevicTT** [comment]
Current changes on branch https://github.com/tenstorrent/tt-metal/compare/pmilojevic/51350-situ

**Comment by kgrujcic-tt** [comment]
Implemented in PR #52448 (branch `kgrujcic/51350-situ`): adds `ttnn.softcap` (Blackhole SFPU unary, the up half) and `ttnn.situ_glu(gate, up, beta1, beta2=None)` (binary composite over pre-split gate/up; `beta2=None` skips the second tanh). `situ_gate` is not exposed standalone.

**Comment by kgrujcic-tt** [comment]
Asked by Filip Vranic to wait for his `tanh` changes before merge, to isolate any potential. Regression. Found 0 issues. Proceeding further with this PR.

**Comment by pmilojevicTT** [comment]
Initial PR for this was merged https://github.com/tenstorrent/tt-metal/pull/52448 currently working on potential improvement for SiTU https://github.com/tenstorrent/tt-metal/issues/52753
